en/net/advanced-operations/securing-and-signing/sign-pdf-document-from-smart-card/_index.md

```diff
@@ -86,48 +86,65 @@ Following are the code snippets to sign a PDF document from a smart card:
 ## Sign With Smart Card Using Signature Field
 
 ```csharp
-// For complete examples and data files, please go to https://github.com/aspose-pdf/Aspose.PDF-for-.NET
-// The path to the documents directory.
-string dataDir = RunExamples.GetDataDir_AsposePdf_SecuritySignatures();
+// For complete examples and data files, check for https://github.com/aspose-pdf/Aspose.PDF-for-.NET
 
-File.Copy(dataDir + "blank.pdf", dataDir + "externalSignature1.pdf", true);
-using (FileStream fs = new FileStream(dataDir + "externalSignature1.pdf", FileMode.Open, FileAccess.ReadWrite))
+private static void GetSignatureInfo()
 {
-    using (Document document = new Document(fs))
+    // The path to the documents directory
+    string dataDir = RunExamples.GetDataDir_AsposePdf_SecuritySignatures();
+    
+    // Open a document stream.
+    using (var fs = new FileStream(dataDir + "blank.pdf", FileMode.Open, FileAccess.ReadWrite))
     {
-        SignatureField field1 = new SignatureField(doc.Pages[1], new Rectangle(100, 400, 10, 10));
-
-        // Sign with certificate selection in the windows certificate store
-        System.Security.Cryptography.X509Certificates.X509Store store = new System.Security.Cryptography.X509Certificates.X509Store(System.Security.Cryptography.X509Certificates.StoreLocation.CurrentUser);
-        store.Open(System.Security.Cryptography.X509Certificates.OpenFlags.ReadOnly);
-        // Manually chose the certificate in the store
-        System.Security.Cryptography.X509Certificates.X509Certificate2Collection sel = System.Security.Cryptography.X509Certificates.X509Certificate2UI.SelectFromCollection(store.Certificates, null, null, System.Security.Cryptography.X509Certificates.X509SelectionFlag.SingleSelection);
-
-        ExternalSignature externalSignature = new ExternalSignature(sel[0])
+        // Open the document from stream
+        using (var document = new Aspose.Pdf.Document(fs))
         {
-            Authority = "Me",
-            Reason = "Reason",
-            ContactInfo = "Contact"
-        };
+            // Create a signature field
+            var field1 = new Aspose.Pdf.Forms.SignatureField(document.Pages[1], new Rectangle(100, 400, 10, 10));
+
+            // Sign with certificate selection in the windows certificate store
+            X509Store store = new X509Store(StoreLocation.CurrentUser);
```

Comment:
We may not specify full qualified name for the System libraries. However, `System.Security.Cryptography.X509Certificates` is enough hard case to specify it.
Additionally var  can be used.